autogen/agentchat/conversable_agent.py

```diff
@@ -1914,6 +1914,16 @@ def _decorator(func: F) -> F:
 
         return _decorator
 
+    def register_model_client(self, model: str, model_client_cls: Client, **kwargs):
```

Comment:
I still don't get why we want to have "model" as a required arg in the signature. Can't a user register for all clients of the same api_type?

Reply:
You need some sort of characteristic to look for in the config file to pull out the config you want to register a client for, unless we want to keep it completely open and have the user specify whatever config filter they want, what were you thinking?

Reply:
@sonichi we can use `api_key`as well, but it would be another argument so it will increase complexity. `model` seems be more commonly used and understood then `api_type`. We think for the first iteration of custom client support we give to the user, we want something that is simple -- it may be limited in functionalities, which we can improve later.

Reply:
I thought that `api_type` needs to be equal to the class name so we can register a class by checking the `api_type` without extra kwargs. If it's not the case, then what is `api_type` used for?

Reply:
I'm revisiting the registration along with the placeholder class, I'll ping you when that's done

Reply:
I have been thinking about this registration thing a bit more. There  is currently several things that is a little confusing here: 
1. api_type can mean “openai” “azure” or a class name
2. Model can be official open ai strings or arbitrary user provided string

so each key now has dual meaning. This is why I was initially confused.

Reply:
`model` has been removed you only register via the `api_type` and don't have to have `model` in the config if you dont' want to